qutebrowser/browser/webkit/network/httpproxy.py

```diff
@@ -0,0 +1,123 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2014-2016 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Asynchronous resolving of HTTP proxies."""
+
+from PyQt5.QtCore import QObject, QEventLoop, pyqtSlot
+from PyQt5.QtNetwork import QNetworkProxy, QNetworkRequest, \
+    QNetworkReply, QNetworkAccessManager
+
+from qutebrowser.browser.webkit.network.pac import PACResolver, EvalProxyError
+from qutebrowser.utils import log
+
+
+def proxy_from_url(typ, url):
+    """Create a QNetworkProxy from QUrl and a proxy type.
+
+    Args:
+        typ: QNetworkProxy::ProxyType.
+        url: URL of a proxy (possibly with credentials).
+
+    Return:
+        New QNetworkProxy.
+    """
+    proxy = QNetworkProxy(typ, url.host())
+    if url.port() != -1:
+        proxy.setPort(url.port())
+    if url.userName():
+        proxy.setUser(url.userName())
+    if url.password():
+        proxy.setPassword(url.password())
+    return proxy
+
+
+class HTTPProxyResolver(QObject):
+
+    """Asynchronous resolver of HTTP proxies."""
+
+    def __init__(self, url):
+        """Resolve a HTTP proxy from URL.
+
+        Args:
+            url: QUrl of a HTTP proxy.
+        """
+        QObject.__init__(self)
```

Comment:
This should be `super().__init__(parent)`, and `__init__` should take a `parent=None` argument (so a parent is passed to Qt correctly when one is passed when instantiating a `HTTPProxyResolver`)